source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1

```diff
@@ -0,0 +1,425 @@
+
+    .PARAMETER ClientReportFailureChecked
+        Indicates whether a client failure is checked.
+
+    .PARAMETER ClientReportType
+        Specifies a client report type. The acceptable values are:
+
+        AllMilestones, AllMilestonesAndAllDetails, ErrorAndWarningMilestones, ErrorMilestones
+
+    .PARAMETER ServerLogChecked
+        Indicates whether a server log is checked.
+
+    .PARAMETER ServerLogFailureChecked
+        Indicates whether a server log failure is checked.
+
+    .PARAMETER ServerLogType
+        Specifies a server log type. The acceptable values are:
+
+        AllMilestones, AllMilestonesAndAllDetails, ErrorAndWarningMilestones, ErrorMilestones
+
+    .PARAMETER ServerReportChecked
+        Indicates whether a server report is checked.
+
+    .PARAMETER ServerReportFailureChecked
+        Indicates whether a server report failure is checked.
+
+    .PARAMETER ServerReportType
+        Specifies a server report type. The acceptable values are:
+
+        AllMilestones, AllMilestonesAndAllDetails, ErrorAndWarningMilestones, ErrorMilestones
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter()]
+        [Boolean]
+        $ClientLogChecked,
+
+        [Parameter()]
+        [Boolean]
+        $ClientLogFailureChecked,
+
+        [Parameter()]
+        [ValidateSet('AllMilestones','AllMilestonesAndAllDetails','ErrorAndWarningMilestones','ErrorMilestones')]
+        [String]
+        $ClientLogType,
+
+        [Parameter()]
+        [Boolean]
+        $ClientReportChecked,
+
+        [Parameter()]
+        [Boolean]
+        $ClientReportFailureChecked,
+
+        [Parameter()]
+        [ValidateSet('AllMilestones','AllMilestonesAndAllDetails','ErrorAndWarningMilestones','ErrorMilestones')]
+        [String]
+        $ClientReportType,
+
+        [Parameter()]
+        [Boolean]
+        $ServerLogChecked,
+
+        [Parameter()]
+        [Boolean]
+        $ServerLogFailureChecked,
+
+        [Parameter()]
+        [ValidateSet('AllMilestones','AllMilestonesAndAllDetails','ErrorAndWarningMilestones','ErrorMilestones')]
+        [String]
+        $ServerLogType,
+
+        [Parameter()]
+        [Boolean]
+        $ServerReportChecked,
+
+        [Parameter()]
+        [Boolean]
+        $ServerReportFailureChecked,
+
+        [Parameter()]
+        [ValidateSet('AllMilestones','AllMilestonesAndAllDetails','ErrorAndWarningMilestones','ErrorMilestones')]
+        [String]
+        $ServerReportType
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode
+
+    try
+    {
+        if ((($state.ClientLogChecked -eq $false) -and ($null -eq $ClientLogChecked) -and ($ClientLogType -or $ClientLogFailureChecked)) -or (($ClientLogChecked -eq $false) -and ($clientLogType -or $ClientLogFailureChecked)))
```

Comment:
check this value,  this is a boolean and would default to false

Reply:
Simplified throw statements